### PR TITLE
Migrate ios_deploy to null safety

### PR DESCRIPTION
**Pull request by jmagman**
Part of #71511

### CODE REVIEW

File: packages/flutter_tools/lib/src/ios/ios_deploy.dart [hunk 1]
```diff
@@ -87,11 +85,11 @@ class IOSDeploy {
...
-    @required String deviceId,
-    @required String bundlePath,
-    @required Directory appDeltaDirectory,
-    @required List<String>launchArguments,
-    @required IOSDeviceConnectionInterface interfaceType,
+    required String deviceId,
+    required String bundlePath,
+    required List<String>launchArguments,
+    required IOSDeviceConnectionInterface interfaceType,
+    Directory? appDeltaDirectory,
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/ios/ios_deploy.dart:92]: is this not always passed? or is it always passed, but sometimes null?

File: packages/flutter_tools/lib/src/ios/ios_deploy.dart [hunk 2]
```diff
@@ -306,8 +304,8 @@ class IOSDeployDebugger {
         _launchCommand,
         environment: _iosDeployEnv,
       );
-      String lastLineFromDebugger;
-      final StreamSubscription<String> stdoutSubscription = _iosDeployProcess.stdout
+      String? lastLineFromDebugger;
+      final StreamSubscription<String>? stdoutSubscription = _iosDeployProcess?.stdout
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/ios/ios_deploy.dart:308]: is `_iosDeployProcess` being null at this point a valid state?

File: packages/flutter_tools/lib/src/ios/ios_deploy.dart [hunk 3]
```diff
@@ -356,18 +354,18 @@ class IOSDeployDebugger {
         }
         lastLineFromDebugger = line;
       });
-      final StreamSubscription<String> stderrSubscription = _iosDeployProcess.stderr
+      final StreamSubscription<String>? stderrSubscription = _iosDeployProcess?.stderr
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/ios/ios_deploy.dart:357]: same question here

File: packages/flutter_tools/lib/src/ios/ios_deploy.dart (hunk 2 above)
Comment (christopherfujino) [reply packages/flutter_tools/lib/src/ios/ios_deploy.dart:308]: from looking at the code, I think this can be `_iosDeployProcess!.stdout`

Comment (christopherfujino) [reply packages/flutter_tools/lib/src/ios/ios_deploy.dart:308]: and i think `stdoutSubscription` can then be non-nullable

File: packages/flutter_tools/lib/src/ios/ios_deploy.dart (hunk 1 above)
Comment (jmagman) [reply packages/flutter_tools/lib/src/ios/ios_deploy.dart:92]: Always passed, can be null.
https://github.com/flutter/flutter/blob/e028d0f046c20c61b7c7c8a23bc456f337410658/packages/flutter_tools/lib/src/ios/application_package.dart#L104

File: packages/flutter_tools/lib/src/ios/ios_deploy.dart (hunk 2 above)
Comment (jmagman) [reply packages/flutter_tools/lib/src/ios/ios_deploy.dart]: Done.

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**